include/range/v3/view/single.hpp

```diff
@@ -38,16 +40,16 @@ namespace ranges
         {
         private:
             friend struct range_access;
-            Val value_;
+            semiregular_t<Val> value_;
             struct cursor
             {
             private:
-                Val value_;
+                semiregular_t<Val> value_;
                 bool done_;
             public:
                 cursor() = default;
-                explicit cursor(Val value)
-                  : value_(std::move(value)), done_(false)
+                explicit cursor(semiregular_t<Val> value_opt)
+                  : value_{detail::move(value_opt)}, done_{false}
```

Comment:
By the time you create a `cursor`, you better have an actual value! IMO, these changes to `cursor` should be reverted.